Documentation/design-docs/DotNetCore-SharedPackageStore.md

```diff
@@ -0,0 +1,218 @@
+  </ItemGroup>
+</Project>
+```
+
+
+and issue a command like below:
+
+```
+dotnet cache --entries packages.xml --framework netcoreapp2.0 [--output C:\Foo] --runtime win7-x64 --fx-version 2.0.0-preview2-00001 [--no-optimize]
+
+--framework          Specifies the TFM that the package store is applicable to
+--output       The output directory to create the package store in (default: %USERPROFILE%\.dotnet or ~/.dotnet)
+
+--no-optimize  Do not perform crossgen of the assemblies after "restore" (optimize is the default)
+
+--runtime          The runtime identifier of the target platform where these assemblies will be run
+--fx-version   The Microsoft.NETCore.App package version that will be used to run the assemblies
+
+```
+
+Hosting providers would create a `packages.xml` file corresponding to the packages that will be shared in their hosting environment and specify the file to `dotnet cache`. The file can be on the file system or from an URL. The TFM argument is used in the shared package layout described above.
+
+If `--optimize` is specified, we would precompile all the managed assets to native code in a temp folder before copying to the output folder. If crossgen is used, it would be the one acquired in the closure of the `Microsoft.NETCore.App` specified by the `--fx-version` option. Also, if no `--output` folder is specified, then the default is `~/.dotnet` or `%USERPROFILE%\.dotnet\`. The output asset files will be present in the following layout: `$HOME/.dotnet/packages/{tfm}/{package-name}/{package-version}/{asset-path}`.
+
+The output folder will be consumed by the runtime by adding to the `DOTNET_SHARED_PACKAGES_ROOTS` environment variable. See probe precedence below.
+
+# Building apps with shared packages
+
+The current mechanism to build applications that share assemblies is by not specifying a RID in the project file. Then, a portable app model is assumed and assemblies that are part of Microsoft.NETCore.App are found under the `dotnet` install root. With shared package store, applications have the ability to filter any set of packages from their publish output. Thus the decision of a portable or a standalone application is not made at the time of project authoring but is instead done at publish time. 
+
+## Project Authoring
+We will by default treat `Microsoft.NETCore.App` as though `type: platform` is always specified, thus requiring no explicit RID specification by the user. It will be an `ERROR` to specify a RID in the csproj file using the `<RuntimeIdentifier/>` tag.
+
+## dotnet restore
+
+Because RIDs are not available until publish time, the application is treated as though it is a present day portable app and a regular restore is performed.
+
+## dotnet build
+
+`dotnet build` should treat any project as though it is a portable app and produce `runtimeconfig.json` with the `Microsoft.NETCore.App` as the framework. In addition, the `runtimeconfig.json` file should also specify the TFM field as:
```

Comment:
>`dotnet build` should treat any project as though it is a portable app

This is not true.  `dotnet build` will build whatever it is told to build.  If someone sets the `RuntimeIdentifier`, it respects that and doesn't build a portable app.